datadog/resource_datadog_service_level_objective.go

```diff
@@ -151,12 +151,110 @@ func ValidateServiceLevelObjectiveTypeString(v interface{}, k string) (ws []stri
 
 func buildServiceLevelObjectiveStruct(d *schema.ResourceData) *datadogV1.ServiceLevelObjective {
 
-	slo := &datadogV1.ServiceLevelObjective{
-		Id:   datadogV1.PtrString(d.Id()),
-		Name: d.Get("name").(string),
-		Type: datadogV1.SLOType(d.Get("type").(string)),
+	slo := datadogV1.NewServiceLevelObjectiveWithDefaults()
+	slo.SetName(d.Get("name").(string))
+	slo.SetType(datadogV1.SLOType(d.Get("type").(string)))
+	slo.SetId(d.Id())
+
+	if attr, ok := d.GetOk("description"); ok {
+		slo.SetDescription(attr.(string))
+	}
+
+	switch slo.GetType() {
+	case datadogV1.SLOTYPE_MONITOR:
+		// monitor type
+		if attr, ok := d.GetOk("monitor_ids"); ok {
+			s := make([]int64, 0)
+			for _, v := range attr.(*schema.Set).List() {
+				s = append(s, int64(v.(int)))
+			}
+			slo.SetMonitorIds(s)
+		}
+		if attr, ok := d.GetOk("groups"); ok {
+			s := make([]string, 0)
+			for _, v := range attr.(*schema.Set).List() {
+				s = append(s, v.(string))
+			}
+			slo.SetGroups(s)
+		}
+	default:
+		// metric type
+		if attr, ok := d.GetOk("query"); ok {
+			queries := make([]map[string]interface{}, 0)
+			raw := attr.([]interface{})
+			for _, rawQuery := range raw {
+				if query, ok := rawQuery.(map[string]interface{}); ok {
+					queries = append(queries, query)
+				}
+			}
+			if len(queries) >= 1 {
+				// only use the first defined query
+				slo.SetQuery(*datadogV1.NewServiceLevelObjectiveQuery(
+					queries[0]["denominator"].(string),
+					queries[0]["numerator"].(string)))
+			}
```

Comment:
I may not fully understand, but since query is defined in the schema as only allowing a MaxItems of `1`, couldn't we just setQuery as d.query and avoid this for loop ?